### PR TITLE
MBS-7257: Add quadratic bezier paths

### PR DESCRIPTION
**Pull request by sh-csg**

### CODE REVIEW

**Review by PhMemmel — COMMENTED** [review]
Review round 1 finished.

Tested manually, works really smooth, good job! That's a very nice extension indeed.

Aside from the comments in the code:
- There is no touch support right now. I am not sure, if this is easily doable, because the paths are very thin? But of course would be nice if an iPad user could use this feature as well.
- Do we need to update playstore version? Probably not, because just the svg path definition is being updated by the editor. The playstore JS object however does not really care if the path is more or less complicated, right?

File: amd/src/learningmap.js [hunk 1]
```diff
@@ -5,6 +5,18 @@ import placestore from 'mod_learningmap/placestore';
...
+    const targetPoints = {
+        firstPoint: 1,
+        secondPoint: 2,
+        bezierPoint: 3,
+    };
+
+    const pathTypes = {
+        line: 1,
+        quadraticbezier: 2,
+    };
```
Comment (PhMemmel) [inline amd/src/learningmap.js:18]: Should these be outside the init method in the module itself?

**Comment by sh-csg** [comment]
Thanks a lot for doing the review!
All addresses issues are fixed. Touch mode is now functional.
The attribute "d" is part of the SVG standard.

**Review by PhMemmel — COMMENTED** [review]
Good job!

Touch mode is now working.

However, there has been a "regression" (not sure if intended). If you "curve" the path and afterwards move the start or end place of it the bezier curve is being subsituted by a straight line again.

File: amd/src/learningmap.js [hunk 2]
```diff
@@ -420,6 +454,75 @@ export const init = () => {
...
+            let parts = oldDefinition.split(' ');
+            let fromX = 0;
+            let fromY = 0;
+            let toX = 0;
+            let toY = 0;
+            let bezierX = 0;
+            let bezierY = 0;
+            let pathType = pathTypes.line;
+
+            // The d attribute of an SVG path in a learning map can have two different formats (in this version):
```
Comment (PhMemmel) [inline amd/src/learningmap.js:476]: Yes, that is exactly the comment needed here, perfect 👍

**Comment by sh-csg** [comment]
Missing "s" in the constant name - fixed also missing constant use in drag()

**Comment by PhMemmel** [comment]
Everything's fine, nice work, integrated!

**Pull request merged**